WearSpeakerSample/wear/src/main/java/com/example/android/wearable/speaker/SpeakerApp.kt

```diff
@@ -88,130 +81,46 @@ fun SpeakerApp() {
         requestPermissionLauncher = rememberLauncherForActivityResult(RequestPermission()) {
             // We ignore the direct result here, since we're going to check anyway.
             scope.launch {
-                lifecycleOwner.lifecycle.withStateAtLeast(Lifecycle.State.STARTED) {
-                    stateHolder.onAction(AppAction.PermissionResultReturned)
-                }
+                mainState.permissionResultReturned()
             }
         }
 
-        // Notify the state holder whenever we become started to reset the state
-        LaunchedEffect(stateHolder) {
+        // Notify the state holder whenever we become stopped to reset the state
+        LaunchedEffect(mainState, scope) {
```

Comment:
Wouldn't be better to add a callback to the `LocalLifecycleOwner`? I think it's easier to understand

Reply:
Replaced with a `DisposableEffect` for setting up a `LifecycleObserver`

Reply:
Thanks! Looks good